### PR TITLE
chore: replace InfracostComment task with `infracost comment`

### PR DESCRIPTION
**Pull request by hugorut**
* Deprecates `InfracostComment` task
* Replaces examples to use `infracost comment`

---
Tested `infracost comment` in pipelines - see pr comment below

### CODE REVIEW

**Comment by hugorut** [comment]
[//]: <> (infracost-comment)

💰 Infracost estimate: **monthly cost will increase by $743 📈**
<table>
  <thead>
    <td>Project</td>
    <td>Previous</td>
    <td>New</td>
    <td>Diff</td>
  </thead>
  <tbody>
    <tr>
      <td>infracost/infracost-azure-devop...mples/terraform-directory/code</td>
      <td align="right">$0</td>
      <td align="right">$743</td>
      <td>+$743</td>
    </tr>
  </tbody>
</table>

<details>
<summary><strong>Infracost output</strong></summary>

```
Project: infracost/infracost-azure-devops/examples/terraform-directory/code

+ aws_instance.web_app
  +$743

    + Instance usage (Linux/UNIX, on-demand, m5.4xlarge)
      +$561

    + root_block_device
    
        + Storage (general purpose SSD, gp2)
          +$5.00

    + ebs_block_device[0]
    
        + Storage (provisioned IOPS SSD, io1)
          +$125
    
        + Provisioned IOPS
          +$52.00

+ aws_lambda_function.hello_world
  Monthly cost depends on usage

    + Requests
      Monthly cost depends on usage
        +$0.20 per 1M requests

    + Duration
      Monthly cost depends on usage
        +$0.0000166667 per GB-seconds

Monthly cost change for infracost/infracost-azure-devops/examples/terraform-directory/code
Amount:  +$743 ($0.00 → $743)

──────────────────────────────────
Key: ~ changed, + added, - removed

2 cloud resources were detected:
∙ 2 were estimated, 2 include usage-based costs, see https://infracost.io/usage-file
```
</details>

This comment will be updated when the cost estimate changes.

<sub>
  Is this comment useful? <a href="https://www.infracost.io/feedback/submit/?value=yes" rel="noopener noreferrer" target="_blank">Yes</a>, <a href="https://www.infracost.io/feedback/submit/?value=no" rel="noopener noreferrer" target="_blank">No</a>
</sub>

File: examples/multi-project/README.md [hunk 1]
```diff
@@ -35,12 +35,14 @@ jobs:
...
           PROD_AWS_SECRET_ACCESS_KEY: $(exampleProdAwsSecretAccessKey)
 
-      - task: InfracostComment@0
-        displayName: Post the comment
-        inputs:
-          githubToken: $(githubToken)
-          path: /tmp/infracost.json
-          behavior: update # Create a single comment and update it. See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
+      - bash: |
+          infracost comment github \
```
Comment (vdmgolub) [inline examples/multi-project/README.md:39]:
```suggestion
          # Create a single comment and update it. See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options
          infracost comment github \
```

File: examples/multi-project/README.md [hunk 2]
```diff
@@ -115,11 +117,14 @@ jobs:
...
-      - task: InfracostComment@0
-        displayName: Post the comment
-        inputs:
-          githubToken: $(githubToken)
-          path: /tmp/infracost_combined.json
-          behavior: update # Create a single comment and update it. See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
+      - bash: |
+          # Create a single comment and update it. See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options
+          infracost comment github \
+          --path /tmp/infracost_combined.json \
```
Comment (vdmgolub) [inline examples/multi-project/README.md:123]:
```suggestion
          --path="infracost_jsons/*.json" \
```

File: examples/multi-project/README.md [hunk 3]
```diff
@@ -115,11 +117,14 @@ jobs:
           infracost output --path="infracost_jsons/*.json" --format=json --out-file=/tmp/infracost_combined.json
```
Comment (vdmgolub) [inline examples/multi-project/README.md:117]: We don't need `output` command anymore. The `infracost comment` can accept the blob pattern. The whole step can be removed.

File: examples/private-terraform-module/README.md [hunk 4]
```diff
@@ -38,11 +38,13 @@ jobs:
...
         # env:
 
-      - task: InfracostComment@0
-        displayName: Post the comment
-        inputs:
-          githubToken: $(githubToken)
-          path: /tmp/infracost.json
-          behavior: update # Create a single comment and update it. See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
+      - bash: |
+          infracost comment github \
```
Comment (vdmgolub) [inline examples/private-terraform-module/README.md:42]:
```suggestion
          # Create a single comment and update it. See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options
          infracost comment github \
```

File: tasks/InfracostComment/task.json [hunk 5]
```diff
@@ -9,9 +9,10 @@
   "author": "Infracost",
   "version": {
     "Major": 0,
-    "Minor": 1,
+    "Minor": 2,
     "Patch": 18
```
Comment (vdmgolub) [inline tasks/InfracostComment/task.json:13]: Should it be 0 as we bump the minor version?
```suggestion
    "Patch": 0
```

File: tasks/InfracostSetup/task.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
   "author": "Infracost",
   "version": {
     "Major": 0,
-    "Minor": 1,
+    "Minor": 2,
     "Patch": 18
```
Comment (vdmgolub) [inline tasks/InfracostSetup/task.json:13]:
```suggestion
    "Patch": 0
```

File: vss-extension.json [hunk 7]
```diff
@@ -2,7 +2,7 @@
   "manifestVersion": 1,
   "id": "infracost-tasks",
   "name": "Infracost Tasks",
-  "version": "0.1.18",
+  "version": "0.2.18",
```
Comment (vdmgolub) [inline vss-extension.json:5]:
```suggestion
  "version": "0.2.0",
```

File: README.md [hunk 8]
```diff
@@ -91,16 +91,15 @@ The Azure Pipelines Infracost tasks can be used with either Azure Repos (only gi
           - bash: infracost breakdown --path=$(TF_ROOT)/plan.json --format=json --out-file=/tmp/infracost.json
             displayName: Run Infracost
 
-          # See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
-          - task: InfracostComment@0
+          # Adds a cost estimate comment to a Azure Repos pull request.
```
Comment (vdmgolub) [inline README.md:94]: To have the same style as other comments:
```suggestion
          # Add a cost estimate comment to a Azure Repos pull request.
```

Also, only line 90 we probably don't need the `Combine Infracost JSON files` comment, as `infracost comment` can do it automatically.

File: README.md [hunk 9]
```diff
@@ -190,19 +189,15 @@ If there are issues, you can enable the 'Enable system diagnostics' check box wh
...
-                githubToken: $(githubToken) # Required to post comments, created in the next step
-                path: /tmp/infracost.json
-                # Choose the commenting behavior, 'update' is a good default:
-                behavior: update # Create a single comment and update it. The "quietest" option.
-                # behavior: delete-and-new # Delete previous comments and create a new one.
-                # behavior: hide-and-new # Minimize previous comments and create a new one.
-                # behavior: new # Create a new cost estimate comment on every push.
-                # Limit the object that should be commented on, either merge-request or commit
-                # targetType: pull-request
+          # Adds a cost estimate comment to a GitHub pull request.
```
Comment (vdmgolub) [inline README.md:192]: To have the same style as other comments:
```suggestion
          # Add a cost estimate comment to a GitHub pull request.
```

Also, only line 188 we probably don't need the `Combine Infracost JSON files` comment, as `infracost comment` can do it automatically.

File: README.md [hunk 10]
```diff
@@ -277,6 +272,10 @@ It accepts the following inputs:
 
 ### InfracostComment
 
+> **THIS TASK IS DEPRECATED**
+>
+> This task will soon be removed, please use `infracost comment` directly.
+
 This task adds comments to GitHub pull requests and commits, or Azure Repos pull requests.
 
 ```yml
```
Comment (vdmgolub) [inline README.md:281]: In troubleshooting section along with `InfracostComment` task let's also mention "`InfracostComment`/`infracost comment`" https://github.com/infracost/infracost-azure-devops/tree/45614559dbf7fac58fb89c6c7cf2646104098869#troubleshooting

File: examples/multi-project/README.md (hunk 3 above)
Comment (hugorut) [reply examples/multi-project/README.md:117]: 👍

File: tasks/InfracostComment/task.json (hunk 5 above)
Comment (hugorut) [reply tasks/InfracostComment/task.json:13]: 👍

File: README.md (hunk 10 above)
Comment (hugorut) [reply README.md:283]: https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: examples/multi-project/README.md (hunk 1 above)
Comment (hugorut) [reply examples/multi-project/README.md:40]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: examples/private-terraform-module/README.md (hunk 4 above)
Comment (hugorut) [reply examples/private-terraform-module/README.md:43]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: examples/multi-project/README.md (hunk 2 above)
Comment (hugorut) [reply examples/multi-project/README.md]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: examples/multi-project/README.md (hunk 3 above)
Comment (hugorut) [reply examples/multi-project/README.md]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: tasks/InfracostComment/task.json (hunk 5 above)
Comment (hugorut) [reply tasks/InfracostComment/task.json]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: tasks/InfracostSetup/task.json (hunk 6 above)
Comment (hugorut) [reply tasks/InfracostSetup/task.json]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: README.md (hunk 8 above)
Comment (hugorut) [reply README.md]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: README.md (hunk 9 above)
Comment (hugorut) [reply README.md]: done in https://github.com/infracost/infracost-azure-devops/pull/45/commits/51e7994d795d728acc2fcd1bef0007f8f0344df2

File: examples/multi-project/README.md (hunk 3 above)
Comment (hugorut) [reply examples/multi-project/README.md]: I'm getting test failures with this though: https://dev.azure.com/infracost/base/_build/results?buildId=1056&view=logs&j=4153937c-ab17-5c86-5eb6-7b0ab0c53920&t=a66737fe-e920-5002-ef61-aac67ed1ea1e

Comment (hugorut) [reply examples/multi-project/README.md]: is this expected, it looks like it's not picking up all the project outputs in `infracost comment`

File: .azure/pipelines/examples-test.yml [hunk 11]
```diff
@@ -127,19 +130,18 @@ jobs:
...
-      - task: InfracostComment@0
-        displayName: Post the comment
-        inputs:
-          githubToken: $(githubToken)
-          path: /tmp/infracost_combined.json
-          behavior: update
-          dryRun: true
+      - bash: |-
+          infracost comment github \
+          --path infracost_jsons/*.json \
```
Comment (vdmgolub) [inline .azure/pipelines/examples-test.yml:135]:
```suggestion
          --path "infracost_jsons/*.json" \
```

This causes the failed matrix test.

File: examples/multi-project/README.md [hunk 12]
```diff
@@ -112,14 +115,13 @@ jobs:
...
-
-      - task: InfracostComment@0
-        displayName: Post the comment
-        inputs:
-          githubToken: $(githubToken)
-          path: /tmp/infracost_combined.json
-          behavior: update # Create a single comment and update it. See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
+          # Create a single comment and update it. See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options
+          infracost comment github \
+          --path infracost_jsons/*.json \
```
Comment (vdmgolub) [inline examples/multi-project/README.md:120]:
```suggestion
          --path "infracost_jsons/*.json" \
```

Sorry, put the comment in the wrong place. This is the CI fix, need to regenerate the examples.

File: scripts/generateExamplesTests.js [hunk 13]
```diff
@@ -91,7 +98,7 @@ function fixupExamples(examples) {
 
           steps.push(
             {
-              bash: `diff ${goldenFilePath} /tmp/slack-message.json`,
+              bash: `diff -y <(jq --sort-keys . ${goldenFilePath}) <(jq --sort-keys . /tmp/slack-message.json)`,
```
Comment (vdmgolub) [inline scripts/generateExamplesTests.js:101]: BTW, this is cool, the diff looks much better 👍

File: README.md [hunk 14]
```diff
@@ -87,20 +87,18 @@ The Azure Pipelines Infracost tasks can be used with either Azure Repos (only gi
...
             displayName: Run Infracost
 
-          # See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
-          - task: InfracostComment@0
+          # Add a cost estimate comment to a Azure Repos pull request.
+          - bash: |
+              infracost comment azure-repos \
+                 --path /tmp/infracost.json \
+                 --azure-repos-token $(System.AccessToken) \
+                 --pull-request $(System.PullRequest.PullRequestNumber) \
```
Comment (vdmgolub) [inline README.md:98]: Azure Repos' PR number stored in a different env var:
```suggestion
                 --pull-request $(System.PullRequest.PullRequestId) \
```

File: README.md [hunk 15]
```diff
@@ -87,20 +87,18 @@ The Azure Pipelines Infracost tasks can be used with either Azure Repos (only gi
...
           - bash: infracost breakdown --path=$(TF_ROOT)/plan.json --format=json --out-file=/tmp/infracost.json
             displayName: Run Infracost
 
-          # See https://github.com/infracost/infracost-azure-devops#infracostcomment for other options
-          - task: InfracostComment@0
+          # Add a cost estimate comment to a Azure Repos pull request.
+          - bash: |
+              infracost comment azure-repos \
+                 --path /tmp/infracost.json \
+                 --azure-repos-token $(System.AccessToken) \
```
Comment (vdmgolub) [inline README.md:97]: My bad, missed that too:
```suggestion
                 --azure-access-token $(System.AccessToken) \
```

File: examples/multi-project/README.md (hunk 12 above)
Comment (hugorut) [reply examples/multi-project/README.md]: nice! fixed in https://github.com/infracost/infracost-azure-devops/pull/45/commits/4fa1b8f25a406aab41ae47519cc8f587351957d1

File: README.md (hunk 15 above)
Comment (hugorut) [reply README.md]: 👍 , fixed in https://github.com/infracost/infracost-azure-devops/pull/45/commits/4fa1b8f25a406aab41ae47519cc8f587351957d1

**Review by vdmgolub — APPROVED** [review]
👍

**Comment by hugorut** [comment]
@vdmgolub thank you for you solid reviews as always ❤️

**Pull request merged**